tests/pipeline/test_refresh_modes.py

```diff
@@ -0,0 +1,266 @@
+from unittest import mock
+
+import pytest
+import dlt
+from dlt.common.pipeline import resource_state
+from dlt.destinations.exceptions import DatabaseUndefinedRelation
+from dlt.destinations.impl.duckdb.sql_client import DuckDbSqlClient
+from dlt.pipeline.state_sync import load_pipeline_state_from_destination
+
+from tests.utils import clean_test_storage, preserve_environ
+from tests.pipeline.utils import assert_load_info
+
+
+def test_refresh_drop_dataset():
```

Comment:
I think it is possible to test also filesystem? we have now a client that can read table data. it would be cool to test local filesystem here